### PR TITLE
feat: Allow user scope to view old events

### PR DESCRIPTION
**Pull request by joonatanaatos**

### CODE REVIEW

File: packages/ilmomasiina-backend/src/models/event.ts [hunk 1]
```diff
@@ -211,20 +210,6 @@ export default function setupEventModel(sequelize: Sequelize) {
...
-                  [Op.gt]: moment().subtract(7, "days").toDate(),
-                },
-                // or happened less than a week ago
-                date: {
-                  [Op.gt]: moment().subtract(7, "days").toDate(),
-                },
-                endDate: {
-                  [Op.gt]: moment().subtract(7, "days").toDate(),
-                },
-              },
```
Comment (PurkkaKoodari) [inline packages/ilmomasiina-backend/src/models/event.ts]: I think we'll still want a hard limit for regular users

File: packages/ilmomasiina-backend/src/routes/events/getEventsList.ts [hunk 2]
```diff
@@ -24,8 +24,20 @@ function eventOrder(): Order {
 export const eventsListForUserCached = createCache({
   maxAgeMs: 1000,
   maxPendingAgeMs: 2000,
-  async get(category?: string) {
-    const where = category ? { category } : {};
+  async get(options: { category?: string; since?: Date }) {
```
Comment (PurkkaKoodari) [inline packages/ilmomasiina-backend/src/routes/events/getEventsList.ts]: This won't work with the current caching system, which compares the function argument with `Object.is` (using a `Map`)

File: packages/ilmomasiina-backend/src/routes/events/getEventsList.ts [hunk 3]
```diff
@@ -24,8 +24,20 @@ function eventOrder(): Order {
...
+    const { category, since } = options;
+    const filters: WhereOptions = {};
+    if (category) {
+      filters.category = category;
+    }
+    if (since) {
+      filters.date = {
+        [Op.gt]: since,
```
Comment (PurkkaKoodari) [inline packages/ilmomasiina-backend/src/routes/events/getEventsList.ts]: `Op.gte` would probably be more exact

File: packages/ilmomasiina-models/src/schema/eventList/index.ts [hunk 4]
```diff
@@ -38,6 +38,11 @@ export const eventListQuery = Type.Object({
       description: "If set, only events with the provided category are included.",
     }),
   ),
+  since: Type.Optional(
+    Type.Date({
+      description: "If set, only events starting after this date are included.",
+    }),
+  ),
```
Comment (PurkkaKoodari) [inline packages/ilmomasiina-models/src/schema/eventList/index.ts]: `Type.Date` is a [typebox extension](https://github.com/sinclairzx81/typebox?tab=readme-ov-file#javascript-types), it doesn't actually parse the dates. You'll need to use `Type.String` with [`format: "date-time"`](https://json-schema.org/draft/2020-12/json-schema-validation#section-7.3.1-4.2) and parse yourself (like we do in e.g. [`updateEvent`](https://github.com/Tietokilta/ilmomasiina/blob/cf0a1fd003f235b1ed0128c38dd0de18e7f03940/packages/ilmomasiina-backend/src/routes/admin/events/updateEvent.ts#L83-L86))

**Review by PurkkaKoodari — CHANGES_REQUESTED** [review]

**Review by kahlstrm — APPROVED** [review]
Looks good 👌

**Pull request merged**